File: Problems/278_First_Bad_Version.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int left  = 1;
        int right = n;
        int res   = 1;
        while(left<=right){
            int middle = (right-left)/2 + left;
            if(isBadVersion(middle)){
                res = middle;
                right = middle-1;
            }else{
                left = middle+1;
            }
        }
        return res;
    }
};

File: Problems/278_First_Bad_Version.py
# The isBadVersion API is already defined for you.
def isBadVersion(version: int) -> bool:
    pass
class Solution:
    def firstBadVersion(self, n: int) -> int:
        left = 1
        right = n
        res = -1
        while left<=right:
            middle = (left+right)//2
            if isBadVersion(middle):
                res = middle
                right = middle-1
            else:
                left = middle+1
        return res